app/models/ext_management_system.rb

```diff
@@ -426,6 +426,32 @@ def enable!
     update!(:enabled => true)
   end
 
+  # override destroy_queue from AsyncDeleteMixin
+  def destroy_queue(deliver_on = Time.now)
+    _log.info("Queuing destroy of #{self.class.name} with id: #{id}")
+    MiqQueue.put(
+      :class_name  => self.class.name,
+      :instance_id => id,
+      :method_name => "orchestrate_destroy",
+      :deliver_on  => deliver_on
+    )
+  end
+
+  # Wait until all associated workers are dead to destroy this ems
+  def orchestrate_destroy
+    disable! if enabled?
+
+    if self.destroy == false
+      raise MiqException::MiqQueueRetryLater.new(:deliver_on => 15.seconds.from_now)
+    end
+  end
+
+  before_destroy :before_destroy
+
+  def before_destroy
```

Comment:
I would opt for a descriptive name, like `assert_no_queues_present` or wrap the assertion in a block

```ruby
before_destroy do |ems|
  throw(:abort) if !MiqWorker.find_current_or_starting.where(:queue_name => ems.queue_name).count.zero?
end
```